### PR TITLE
3.1.x Add additional SAML configurations

### PR DESCRIPTION
**Pull request by edtubillara**
This adds more SAML configurations:
1. Make signing auth requests for SAML
2. Add file source for SAML metadata provider
3. Add the name format for the SAML mapping attributes

### CODE REVIEW

File: roles/keystone/templates/etc/keystone/keystone.conf [hunk 1]
```diff
@@ -109,7 +102,7 @@ remote_id_attribute = {{ sp.remote_id_attribute }}
 
 {% if keystone.federation.enabled|bool  and keystone.federation.sp.saml.enabled|bool -%}
 
-[saml]
+[saml2]
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone.conf]: Can you explain this change?

**Comment by nirajdp76** [comment]
ok to test

File: roles/keystone/templates/etc/keystone/keystone.conf (hunk 1 above)
Comment (edtubillara) [inline roles/keystone/templates/etc/keystone/keystone.conf]: It seems that when you set the auth methods, Keystone is able to already tell which is the mapped object. And setting an auth method to the mapped object is deprecated. So when we put `saml2`, Keystone can associate that to a config section called `saml2`.

**Comment by edtubillara** [comment]
retest

**Comment by lihkin213** [comment]
retest

**Pull request merged**